:memo: Clarify passwords with databases (#2799)
platformsh/platformsh-docs: docs/src/add-services/mysql/_index.md

For further details, see the [MariaDB documentation](https://mariadb.com/kb/en/character-set-and-collation-overview/).

## Password generation
 
When you create an endpoint to allow your app and database to communicate,
an empty password is generated by default.
If you add multiple endpoints,
having empty passwords can cause technical issues with your app.
 
To solve this,
define custom endpoints in your [service configuration](#configuration-options),
similarly to when you create [multiple databases](#multiple-databases).
For each custom endpoint you create, 

> you get an automatically generated password.

Author: I did move things around for clarity while making sure my sentences remained as short as possible.

After your custom endpoints are exposed as relationships in your [app configuration](../../create-apps/_index.md),
you can retrieve the password for each endpoint

> through the `PLATFORM_RELATIONSHIPS` [environment variable](../../development/variables/use-variables.md#use-platformsh-provided-variables).

Author: I left the first link unchanged because: 
- Before users can access the environment variable, imho I believe it's best if they at least have a rough idea of what it is.
- I added another sentence to explain how they can access the variable with links to both the shell method and the Config Reader method. I believe both sentences complement each other, what do you think?

> through the `PLATFORM_RELATIONSHIPS` [environment variable](../../development/variables/use-variables.md#use-platformsh-provided-variables).

Reviewer: Yeah that definitely makes sense! Context first and then how to action that knowledge.

For further details, see the [MariaDB documentation](https://mariadb.com/kb/en/character-set-and-collation-overview/).

## Password generation
 
When you create an endpoint to allow your app and database to communicate,

> an empty password is generated by default.

Reviewer 2: :question: I thought that the empty password was when you didn't specify an endpoint. Is that not true? It was my impression from the Context that you could generate a password specifically by defining an endpoint.

If you add multiple endpoints,

> having empty passwords can cause issues with your app.

Reviewer 2: :lady_beetle: This seems to imply that you can add multiple endpoints and have empty passwords. That isn't something that can happen. It's not a problem that _users_ need to resolve, as implied in the next line.

 
To solve this,
define custom endpoints in your [service configuration](#configuration-options).
For each custom endpoint you create, 
you get an automatically generated password,
similarly to when you create [multiple databases](#multiple-databases).
Note that you can't customize these automatically generated passwords.

After your custom endpoints are exposed as relationships in your [app configuration](../../create-apps/_index.md),
you can retrieve the password for each endpoint
through the `PLATFORM_RELATIONSHIPS` [environment variable](../../development/variables/use-variables.md#use-platformsh-provided-variables).

> You can access the `PLATFORM_RELATIONSHIPS` environment variable [in a shell](../../development/variables/use-variables.md#access-variables-in-a-shell)

Reviewer 2: :question: Why would you want to do this in this case? What's the benefit of pushing people toward this option as the first one?

> or directly [in your app](../../development/variables/use-variables.md#access-variables-in-your-app), for example using the Platform.sh Configuration Reader.

Reviewer 2: :clipboard: 
```suggestion
or directly [in your app](../../development/variables/use-variables.md#access-variables-in-your-app),
for example using a Platform.sh Config Reader.
```
